Source/ReorderController.swift

```diff
@@ -57,6 +57,14 @@ public protocol TableViewReorderDelegate: class {
      */
     func tableView(_ tableView: UITableView, canReorderRowAt indexPath: IndexPath) -> Bool
     
+    /**
+     When attempting to move a row from a sourceIndexPath to a proposedDestinationIndexPath, asks the reorder delegate what the actual targetIndexPath should be. This allows the reorder delegate to selectively allow or modify reordering between sections or groups of rows, for example.
+     - Parameter tableView: The table view requesting this information.
+     - Parameter sourceIndexPath: The original index path of the row to be moved.
+     - Parameter proposedDestinationIndexPath: The potential index path of the row's new location.
+     */
+    func tableView(_ tableView: UITableView, targetIndexPathForMoveFromRowAt sourceIndexPath: IndexPath, to proposedDestinationIndexPath: IndexPath) -> IndexPath
```

Comment:
For consistency with the other delegate methods, this should be named `tableView(_:targetIndexPathForReorderFromRowAt:to:)`.